最新メッセージを自動更新で表示する

yukri1030 opened the pull request:
＃　WHAT
自動更新のため、jsファイルやjbuilderの実装を行なった
message.js ...  ajaxによる最新メッセージの取得、doneメソッドによる表示の実装
index.json.jbuilder....最新のメッセージを配列で取得　変数の定義


# WHY
最新のメッセージを見やすく表示し、より便利にchat-spaceを使っていくため

https://gyazo.com/e6cd87663056ea78aba53b4ab27b350f

https://gyazo.com/9c6e9ed185adc9387e4f0c590cd9f071

yukri1030 committed:
apiのルートを設定

# WHAT
自動更新を行うため、apiのルートを設定した

yukri1030 committed:
apiのコントローラを設定

# WHAT
apiのコントローラーを設定し,ajax で送信した、最新のメッセージidを受け取れるようにした。
respond_toでフォームを分けた

yukri1030 committed:
不要なコメントアウトを削除した

yukri1030 committed:
apiフォルダのmessageコントローターの作成

# WHAT
api/messages_controller.rbの作成
jsonでも受け取れるように、viewを分けた

# WHY
自動更新実装のため

yukri1030 committed:
自動更新のための実装

# WHAT
jsファイルより、ajaxで最新のメッセージidを送り、done メソッドで受け取って表示できるまで行なった

# WHY
最新のメッセージ表示のため

yukri1030 committed:
メッセージに、カスタムデータ属性を持たせる

# WHAT
自動更新実装のため、メッセージ一つ一つに、カスタム属性の、message.idとgroup.idを持たせることした

＃WHY
より便利な機能を使えるようにするため

yukri1030 committed:
カスタムデータ属性を持たせる

# WHAT
message.idのカスタムデータ属性を持たせることとした。

# WHY
より便利な機能（自動更新）の実装のため

yukri1030 committed:
カスタム属性を持たせる

＃WHAT
変数message一つ一つに、カスタム属性である、message.idとgroup.idを定義した。

# WHY
最新のメッセージを見やすく表示するため

i-date reviewed (approved):
ご提出ありがとうございます！
拝見しましたが、特に問題ありません！LGTMとさせていただきます^^

yukri1030 commented:
ありがとうございます！他、カレントユーザーの表示やビューの崩れなど直したら、マージしたいと思っています！

yukri1030 committed:
不要な記述を削除した

＃WHA
カレントユーザーが2名表示されているため
 @group.users << current_user
の記述を削除した

editアクションの@group.usersの記述も不要のため、削除した

# WHY
より見やすい表示のため

yukri1030 committed:
リンクのついているグループ名に色をつけた

リンクのついている、グループ名の文字に、色をつけた

scssを変えた

yukri1030 committed:
ユーザーの表示の部分のscssを直した

ユーザー表示部分のscssを直した
高さやpaddingの調整
